YarpRobotLoggerDevice building error related to `opencv`

Installing with enabling the `YarpRobotLoggerDevice` gives the following error:

```
In file included from /home/hosameldin/code/robotology-superbuild/src/bipedal-locomotion-framework/devices/YarpRobotLoggerDevice/src/YarpRobotLoggerDevice.cpp:24:
/home/hosameldin/code/robotology-superbuild/src/bipedal-locomotion-framework/devices/YarpRobotLoggerDevice/include/BipedalLocomotion/YarpRobotLoggerDevice.h:17:10: fatal error: opencv2/opencv.hpp: No such file or directory
   17 | #include <opencv2/opencv.hpp>
      |          ^~~~~~~~~~~~~~~~~~~~
compilation terminated.
make[2]: *** [devices/YarpRobotLoggerDevice/CMakeFiles/YarpRobotLoggerDevice.dir/build.make:76: devices/YarpRobotLoggerDevice/CMakeFiles/YarpRobotLoggerDevice.dir/src/YarpRobotLoggerDevice.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:1727: devices/YarpRobotLoggerDevice/CMakeFiles/YarpRobotLoggerDevice.dir/all] Error 2
make: *** [Makefile:136: all] Error 2
```

I was on the release [`0.7.0`](https://github.com/ami-iit/bipedal-locomotion-framework/releases/tag/v0.7.0) then switched to @prashanthr05 's branch which is based on current `master`.

According to @prashanthr05, `opencv` is linked properly only if we set `FRAMEWORK_USE_PCL` to true. Actually when setting the option to true (and installing the dependency), the installation goes without error!

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -5,6 +5,7 @@ All notable changes to this project are documented in this file.
 ### Added
 ### Changed
 ### Fix
+- Fix the dependency required to compile the YarpRobotLogger device (https://github.com/ami-iit/bipedal-locomotion-framework/pull/554)
 
 ## [0.8.0] - 2022-07-29
 ### Added

--- a/cmake/BipedalLocomotionFrameworkDependencies.cmake
+++ b/cmake/BipedalLocomotionFrameworkDependencies.cmake
@@ -222,7 +222,7 @@ framework_dependent_option(FRAMEWORK_COMPILE_CalibrationDeltaUpdaterApplication
 
 framework_dependent_option(FRAMEWORK_COMPILE_YarpRobotLoggerDevice
   "Do you want to generate and compile the YarpRobotLoggerDevice?" ON
-  "FRAMEWORK_COMPILE_RobotInterface;FRAMEWORK_COMPILE_YarpImplementation;FRAMEWORK_COMPILE_YarpUtilities;FRAMEWORK_USE_robometry" OFF)
+  "FRAMEWORK_COMPILE_RobotInterface;FRAMEWORK_COMPILE_YarpImplementation;FRAMEWORK_COMPILE_Perception;FRAMEWORK_COMPILE_YarpUtilities;FRAMEWORK_USE_robometry" OFF)
 
 framework_dependent_option(FRAMEWORK_COMPILE_VectorsCollectionWrapper
   "Do you want to generate and compile the VectorsCollectionWrapper?" ON

